controllers/funding/programmes.js

```diff
@@ -128,27 +136,114 @@ function initProgrammesList(router, config) {
     });
 }
 
-function initProgrammeDetail(router, config) {
-    // Allow for programmes without heroes
-    const defaultHeroImage = createHeroImage({
+function renderProgrammeDetail({ res, entry }) {
+    /**
+     * Allow for programmes without heroes
+     */
+    const defaultProgrammeHeroImage = createHeroImage({
         small: 'hero/working-families-small.jpg',
         medium: 'hero/working-families-medium.jpg',
         large: 'hero/working-families-large.jpg',
         default: 'hero/working-families-medium.jpg'
     });
 
-    router.get('/programmes/:slug', function(req, res) {
+    res.render('pages/funding/programme-detail', {
+        entry: entry,
+        title: entry.title,
+        heroImage: entry.hero || defaultProgrammeHeroImage
+    });
+}
+
+function handleProgrammeDetail(slug) {
+    return function(req, res) {
+        const locale = req.i18n.getLocale();
         contentApi
-            .getFundingProgramme({
-                locale: req.i18n.getLocale(),
-                slug: req.params.slug
+            .getFundingProgramme({ locale, slug })
+            .then(entry => {
+                if (entry.contentSections.length > 0) {
+                    renderProgrammeDetail({ res, entry });
+                } else {
+                    throw new Error('NoContent');
+                }
             })
+            .catch(err => {
+                renderNotFoundWithError(err, req, res);
+            });
+    };
+}
+
+function initProgrammeDetail(router) {
+    router.get('/programmes/:slug', (req, res) => {
+        handleProgrammeDetail(req.params.slug)(req, res);
+    });
+}
+
+function initProgrammeDetailAfaEngland(router, options) {
+    const testFn = ab.test('blf-afa-rollout-england', {
+        cookie: {
+            name: config.get('cookies.abTestAwardsForAll'),
+            maxAge: moment.duration(4, 'weeks').asMilliseconds()
+        },
+        id: 'O9FsbkKfSeOammAP0JrEyA'
```

Comment:
Note to self: It might make more sense to start a new experiment rather than reusing the previous experiment ID. Even though the test is the same in principle the URLs are different so might upset Google Experiments.